pycaret/tests/test_time_series.py

```diff
@@ -2,47 +2,108 @@
+from pycaret.internal.ensemble import _ENSEMBLE_METHODS
 import numpy as np
 import pandas as pd
 
+
 @pytest.fixture(scope="session", name="load_data")
 def load_data():
     """Load Pycaret Airline dataset."""
     from pycaret.datasets import get_data
-    airline_data = get_data("airline")
-    return airline_data
 
+    return get_data("airline")
+
+
+@pytest.fixture(scope="session", name="load_setup")
+def load_setup(load_data):
+    """Create a TimeSeriesExperiment to test module functionalities"""
+    from pycaret.internal.PycaretExperiment import TimeSeriesExperiment
+    import numpy as np
+
+    ts_experiment = TimeSeriesExperiment()
+
+    fh = np.arange(1, 13)
+    fold = 3
+
+    return ts_experiment.setup(
+        data=load_data, fh=fh, fold=fold, fold_strategy="expandingwindow", verbose=False
+    )
+
+
+@pytest.fixture(scope="session", name="load_models")
+def load_ts_models(load_setup):
+    """Load all time series module models"""
+    from pycaret.containers.models.time_series import get_all_model_containers
+
+    globals_dict = {"seed": 0, "n_jobs_param": -1}
+    ts_models = get_all_model_containers(globals_dict)
+    ts_experiment = load_setup
+    ts_estimators = []
+
+    for key in ts_models.keys():
+        if not key.startswith(("ensemble", "poly")):
+            ts_estimators.append(ts_experiment.create_model(key))
 
-models = ['naive', 'poly_trend', 'arima', 'exp_smooth', 'theta']
-parametrize_list = [
-    (choice(models))
-]
+    return ts_estimators
+
+
+models = ["naive", "poly_trend", "arima", "exp_smooth", "theta"]
+parametrize_list = [(choice(models))]
 
 
 @pytest.mark.parametrize("model", parametrize_list)
 def test_create_model(model, load_data):
 
     from pycaret.internal.PycaretExperiment import TimeSeriesExperiment
-    exp = TimeSeriesExperiment()
 
+    exp = TimeSeriesExperiment()
     exp.setup(
-        data=load_data,
-        fold=3,
-        fh=12,
-        fold_strategy="expandingwindow"
+        data=load_data, fold=3, fh=12, fold_strategy="expandingwindow", verbose=False
     )
 
     model_obj = exp.create_model(model)
     y_pred = model_obj.predict()
     assert isinstance(y_pred, pd.Series)
     expected = pd.core.indexes.period.PeriodIndex(
         [
-            '1957-05', '1957-06', '1957-07', '1957-08', '1957-09', '1957-10',
-            '1957-11', '1957-12', '1958-01', '1958-02', '1958-03', '1958-04'
+            "1957-05",
+            "1957-06",
+            "1957-07",
+            "1957-08",
+            "1957-09",
+            "1957-10",
+            "1957-11",
+            "1957-12",
+            "1958-01",
+            "1958-02",
+            "1958-03",
+            "1958-04",
         ],
-        dtype='period[M]',
-        freq='M'
+        dtype="period[M]",
+        freq="M",
     )
     assert np.all(y_pred.index == expected)
 
+
+@pytest.mark.parametrize("method", _ENSEMBLE_METHODS)
+def test_blend_model(load_setup, load_models, method):
```

Comment:
Could you add a unit test for blending of only a subset of models?

```python
ts_blender = exp.blend_models([arima_model, naive_model], optimize='MAPE_ts')
```

Reply:
@ngupta23 could you clarify some details about what are we expecting to test by pairs?

Reply:
I think in the example notebook you have an example with blending only a subset of models. It would be good to check this through a unit test to make sure that it does not always produce the same results as when all models are blended.

Reply:
sure, @ngupta23 I added a [functional test](https://github.com/pycaret/pycaret/blob/time_series_ensemble_models/pycaret/tests/test_time_series.py#L112) to assert that predictions from `mean`, `median` and `voting` ensemble differ considering all the models available in the ts module.